src/data/store.rs

```diff
@@ -258,36 +260,21 @@ impl Store {
         self.search
             .update_downloaded_file_in_search_results(&file_id, false);
 
-        SignalToUI::set_ui_signal();
         Ok(())
     }
 
-    pub fn process_event_signal(&mut self) {
-        self.update_downloads();
-        self.update_chat_messages();
-        self.update_search_results();
-        self.update_load_model();
-    }
+    pub fn handle_action(&mut self, action: &Action) {
```

Comment:
This replaced `process_event_signal` in spirit.

Instead of polling many data sources, this forwards the action (message) down into other modules so they decide if they need to do something.